app/code/Magento/InventoryAdminUi/Model/Source/SourceCoordinatesDataProcessor.php

```diff
@@ -0,0 +1,31 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryAdminUi\Model\Source;
+
+/**
+ * Prepare source coordinates data (latitude and longitude). Specified for form structure
+ */
+class SourceCoordinatesDataProcessor
+{
+    /**
+     * @param array $data
+     * @return array
+     */
+    public function process(array $data): array
```

Comment:
why do we need this class at all?
all of this should be done in Source model implementation